### PR TITLE
Feat: Refine Footer Component

### LINKED ISSUE #80: Refine Footer Block
- [ ] Make sizing look good on all reasonable screen sizes.
- [ ] Squash icons to the right instead of the current full-spread.
- [ ] Fix broken icons on mobile aspect ratios.

### PR DESCRIPTION
**Pull request by ethanzitting**
Adjust the styling of the footer component to better match the designs.

Fix bugs in the styling of the footer at narrow aspect ratios.

Reformat the footer component to use more Typescript.

closes #80

### CODE REVIEW

**Review by fredlawl — CHANGES_REQUESTED** [review]
I think I saw some linter issues that need to be worked out, otherwise looks pretty good. I left a few comments about adjusting our social urls.

File: src/components/Footer/Footer.tsx [hunk 1]
```diff
@@ -4,52 +4,55 @@ import React from 'react';
...
+            src: '/images/facebook-logo.svg',
+            alt: 'Facebook Logo',
+        },
+        {
+            url: 'https://www.meetup.com/',
+            src: '/images/meetup-logo.svg',
+            alt: 'Meetup Logo',
+        },
+        {
+            url: 'https://discord.com/',
```
[the remark this answers is not in the document]
Comment (ethanzitting) [reply src/components/Footer/Footer.tsx:27]: Good catch, implemented!

File: src/components/Footer/Footer.tsx [hunk 2]
```diff
@@ -4,52 +4,55 @@ import React from 'react';
...
 
 export function Footer(): JSX.Element {
+    const socialIcons = [
+        {
+            url: 'https://www.facebook.com/',
+            src: '/images/facebook-logo.svg',
+            alt: 'Facebook Logo',
+        },
+        {
+            url: 'https://www.meetup.com/',
```
[the remark this answers is not in the document]
Comment (ethanzitting) [reply src/components/Footer/Footer.tsx:22]: Good catch, implemented!

File: src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -4,52 +4,55 @@ import React from 'react';
 import styles from './Footer.module.scss';
 
 export function Footer(): JSX.Element {
+    const socialIcons = [
+        {
+            url: 'https://www.facebook.com/',
+            src: '/images/facebook-logo.svg',
+            alt: 'Facebook Logo',
+        },
```
[the remark this answers is not in the document]
Comment (ethanzitting) [reply src/components/Footer/Footer.tsx:20]: Sounds good.

File: src/components/Footer/Footer.tsx [hunk 4]
```diff
@@ -4,52 +4,55 @@ import React from 'react';
...
+        {
+            url: 'https://discord.com/',
+            src: '/images/discord-logo.svg',
+            alt: 'Discord Logo',
+        },
+        {
+            url: 'https://twitter.com/?lang=en',
+            src: '/images/twitter-logo.svg',
+            alt: 'Twitter Logo',
+        }
```
[the remark this answers is not in the document]
Comment (ethanzitting) [reply src/components/Footer/Footer.tsx]: Sounds Good.

**Review by fredlawl — APPROVED** [review]

**Pull request merged**